plugins/modules/web_infrastructure/jira.py

```diff
@@ -655,10 +673,16 @@ def request(
         headers = {}
         if isinstance(additional_headers, dict):
             headers = additional_headers.copy()
-        headers.update({
-            "Content-Type": content_type,
-            "Authorization": "Basic %s" % auth,
-        })
+        if self.vars.token is not None:
+            headers.update({
+                "Content-Type": content_type,
+                "Authorization": "Bearer %s" % self.vars.token,
+            })
+        else:
+            headers.update({
```

Comment:
I think the `auth` string computation (and its preceding comment) should be moved from line 670 to here.